tests/test_client.py

```diff
@@ -116,3 +119,34 @@ def my_content_type(path):
 
     for suffix, content_type in mimes:
         _test_write_content_type(suffix, content_type, rig)
+
+
+@register_path_class("mys3")
+class MyS3Path(S3Path):
+    cloud_prefix: str = "mys3://"
+
+
+@register_client_class("mys3")
+class MyS3Client(S3Client):
+    pass
```

Comment:
Wondering if we can isolate these in case they cause some weird interaction failure happens in the future, e.g., in a fixture:

```python
@pytest.fixture
def custom_s3_path():
    @register_path_class("mys3")
    class MyS3Path(S3Path):
        cloud_prefix: str = "mys3://"
  
    @register_client_class("mys3")
    class MyS3Client(S3Client):
        pass

    yield

    # cleanup after use
    implementation_registry.pop("mys3")
```

Reply:
Yep! makes sense.

Typing takes a slight hit within the tests (I tried `custom_s3_path: tuple[custom_s3_path.MyS3Path, custom_s3_path.MyS3Client]` which didn't seem to do anything in pyright for me locally), but it works!